### PR TITLE
Random Pill bottle colors

### PR DESCRIPTION
**Pull request by Kryyto**
## About The Pull Request

This adds 6 news Pill Bottle colors while keeping the default one, the colors are randoms.

## Why It's Good For The Game

Add variance of color for the same item.

## Changelog
:cl:
add: Pill Bottle now have a random color.
/:cl:
randompillbottle

### CODE REVIEW

File: code/game/objects/items/storage/firstaid.dm [hunk 1]
```diff
@@ -318,15 +318,21 @@
...
 	w_class = WEIGHT_CLASS_SMALL
+	var/pill_variance = 100 //probability pill_bottle has a different icon state. Put at 0 for no variance
+	var/pill_type = "pill_canister_"
 
 /obj/item/storage/pill_bottle/ComponentInitialize()
+	var/proper_name = name
 	. = ..()
+	name = proper_name
+	if(prob(pill_variance))
+		icon_state = "[pill_type][rand(0,6)]"
```
Comment (PowerfulBacon) [inline code/game/objects/items/storage/firstaid.dm]: ew non component things in comopnentinitialize

Comment (ike709) [reply code/game/objects/items/storage/firstaid.dm]: What bacon means is put these changes in regular Initialize()

Comment (Kryyto) [reply code/game/objects/items/storage/firstaid.dm]: Ah alright, will do

**Comment by Kryyto** [comment]
Alright, that's fixed @PowerfulBacon

File: code/game/objects/items/storage/firstaid.dm [hunk 2]
```diff
@@ -318,12 +318,21 @@
...
 	lefthand_file = 'icons/mob/inhands/equipment/medical_lefthand.dmi'
 	righthand_file = 'icons/mob/inhands/equipment/medical_righthand.dmi'
 	w_class = WEIGHT_CLASS_SMALL
+	var/pill_variance = 100 //probability pill_bottle has a different icon state. Put at 0 for no variance
+	var/pill_type = "pill_canister_"
+
+/obj/item/storage/pill_bottle/Initialize()
+	var/proper_name = name
+	. = ..()
+	name = proper_name
```
Comment (PowerfulBacon) [inline code/game/objects/items/storage/firstaid.dm]: What is the point of this name stuff?

Comment (Kryyto) [reply code/game/objects/items/storage/firstaid.dm]: Won't this change the name of the icon for what the icon become ? I saw that being used somewhere else too

Comment (PowerfulBacon) [reply code/game/objects/items/storage/firstaid.dm]: Name and icon are independant (Unless the icon state is set to do something with the name), shouldn't affect pillbottles. What other thing was it on?
..() shouldn't be changing the name, so it should not be needed here

Comment (Kryyto) [reply code/game/objects/items/storage/firstaid.dm]: Okay, that's what I thought too but I preferred to do as others. It was also used in the asteroid floor tile: 
image

**Comment by Kryyto** [comment]
That's better @PowerfulBacon ? And I hope you'll have good year by the way.

**Review by PowerfulBacon — APPROVED** [review]
Thanks, you too

**Comment by PowerfulBacon** [comment]
ah shit actually it need ..() but I missed that so now pill bottles won't initialize at all

**Comment by PowerfulBacon** [comment]
Also dice bottles are a subtype of pillbottle

**Pull request merged**